src/rgw/driver/sfs/bucket.cc

```diff
@@ -148,149 +147,96 @@ int SFSBucket::list(
     }
   }
 
-  if (list.objects(
+  // Version listing on unversioned buckets is equivalent to object listing
+  const bool want_list_versions =
+      versioning_enabled() ? params.list_versions : false;
+  const bool listing_succeeded =
+      (want_list_versions && list.versions(
+                                 get_bucket_id(), params.prefix, start_with,
+                                 max, results.objs, &results.is_truncated
+                             )) ||
+      list.objects(
```

Comment:
Maybe I'm missing something here, but won't this be running the `list.objects()` even if we `want_list_versions` but fail at `list.versions()` ? Is that meant by design?

Reply:
Good catch. Tired to be clever..